### PR TITLE
add an initial cypress test to verify the list pages

### PR DESCRIPTION
**Pull request by shieldo**
## Description

Starting off a functional end-to-end test suite using Cypress so that regressions can be checked easily.

Just adding an initial one to verify the list pages at first - will add others later.

### CODE REVIEW

File: e2e/integration/list-requests.cy.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+describe('check the lists of requests and donators are there and not empty', () => {
```
Comment (gregdyke) [inline e2e/integration/list-requests.cy.js:1]: while we're at it, separate into 2 tests with better descriptions that it('passes)?

Comment (shieldo) [reply e2e/integration/list-requests.cy.js:1]: Ah yes, that description was auto-generated and I didn't spot it! I'll separate things out better (and describe them better!) in further commits.

**Comment by shieldo** [comment]
I've separated these out and described them better.

**Comment by shieldo** [comment]
Have added a couple of e2e tests for the happy paths for adding a laptop request and a donator.

**Comment by shieldo** [comment]
Merging this one in first, then will add further tests in another PR!

**Pull request merged**